core/project_settings.cpp

```diff
@@ -963,6 +963,41 @@ bool ProjectSettings::has_custom_feature(const String &p_feature) const {
 	return custom_features.has(p_feature);
 }
 
+ProjectSettings::CustomMap ProjectSettings::get_default_settings(const String &project_name, const String &driver) {
+	ProjectSettings::CustomMap initial_settings;
+
+	if (driver == "GLES3") {
+		initial_settings["rendering/quality/driver/driver_name"] = "GLES3";
+	} else {
+		initial_settings["rendering/quality/driver/driver_name"] = "GLES2";
+		initial_settings["rendering/vram_compression/import_etc2"] = false;
+		initial_settings["rendering/vram_compression/import_etc"] = true;
+	}
+	initial_settings["application/config/name"] = project_name;
+	initial_settings["application/config/icon"] = "res://" + get_default_icon_name();
+	initial_settings["rendering/environment/default_environment"] = "res://" + get_default_env_name();
+
+	return initial_settings;
+}
+
+String ProjectSettings::get_default_env_content() {
+	String default_env_str;
+	default_env_str += "[gd_resource type=\"Environment\" load_steps=2 format=2]\n";
+	default_env_str += "[sub_resource type=\"ProceduralSky\" id=1]\n";
+	default_env_str += "[resource]\n";
+	default_env_str += "background_mode = 2\n";
+	default_env_str += "background_sky = SubResource( 1 )\n";
+	return default_env_str;
```

Comment:
I don't think this is the correct approach. This means that if we change anything in the 'real' default environment we now have to change it in two places. This should serialize a normal default environment instead.

Reply:
I just moved the code from project_manager.cpp as it was.
See: https://github.com/godotengine/godot/blob/07498d313c45f54b0a7c6c0b23e635bddf6f0523/editor/project_manager.cpp#L496-L500

Where can I find the default environment configuration?

Reply:
I confirm that it was hard coded before, maybe default environment could be instanced and serialized to file that way?